lib/model/model.go

```diff
@@ -2761,3 +2989,38 @@ func sanitizePath(path string) string {
 
 	return strings.TrimSpace(b.String())
 }
+
+func encTokenPath(cfg config.FolderConfiguration) string {
+	return filepath.Join(cfg.MarkerName, "syncthing-enc_pw_token")
```

Comment:
No specific reason, might also live in db. Should it?

Reply:
Why does it need to be stored anywhere anyway? 


As far as I understand it's just a thing that is used to verify the sides agree on a password/verify the password hasn't changed?

Reply:
Trusted nodes create it on demand from the passwords. However untrusted nodes can't do that and they need to provide it in cluster configs, thus they need to persist it. That's the entire point of the token, if e.g. a trusted device connects to an untrusted one, the untrusted device needs to be able to present the token it got before when another trusted device connects later.

Reply:
I guess this sort of reduces the usefulness of per-device-per-folder passwords, as for two plaintext devices to sync to a third encrypted device, they would have to use the same password.

Sure for a 4th device they could use a different one, which I guess is why we have this, but I wonder if it's worth it.

Reply:
> I guess this sort of reduces the usefulness of per-device-per-folder passwords, as for two plaintext devices to sync to a third encrypted device, they would have to use the same password.

That's not just because of password tokens though, that's intrinsic in the encryption protocol: If two trusted devices did use different passwords for a single untrusted device, they wouldn't sync with each other at all: Due to different keys, they'd generate different encrypted filenames for the same file. The untrusted file would then have two files for every "unencrypted" file and when one trusted device receives such a duplicate file from the other trusted device through the untrusted one, it won't be able to decrypt it.

Whether the possibility to have separate passwords for the same folder for different devices is useful is another matter. I think that was discussed on the forum thread and finally decided to have it.